This script is used to find the average distance between two randomly chosen points in a unit circle.
Inspiration Drawn from this [youtube video](https://www.youtube.com/watch?v=gCgE07fKQpY)

In [1]:
import numpy as np
import random

def gen_point_pair():
    #generates our random point w/i the unit circle
    x = random.uniform(-1,1)
    y =  random.uniform(-(1-x**2),(1-x**2))
    return [x,y]

In [2]:
def find_ave_distance(num_point_pairs):
    #generates point pairs and finds their average distance
    distances = [find_points_distance(gen_point_pair(),gen_point_pair()) for _ in range(num_point_pairs)]
    return np.mean(distances)

In [3]:
import math
def find_points_distance(p1,p2):
    #finds the distance between the passed point pair
    return math.sqrt((p2[0]-p1[0])**2 + ((p2[1]-p1[1])**2))

In [4]:
import matplotlib.pyplot as plt
num_point_pairs = int(input('Enter the # of point pairs you wish to use to simulate '
                        'the average distance between 2 randomly chosen points in a unit circle'))

print(f'Average Distance using {num_point_pairs} points for simulation: {find_ave_distance(num_point_pairs)}')

"""
#making graph of simulations using a number of point pars up to user bound
distance_approxes = [find_ave_distance(num_p_pairs) for num_p_pairs in range(1,num_point_pairs+1)]
plt.plot(np.arange(num_point_pairs),distance_approxes)
plt.title('Average Distance in Unit Circle Simulations')
plt.xlabel('# of points used in simulation')
plt.ylabel('Average Distance')
plt.show()
"""

Average Distance using 10000000 points for simulation: 0.9089196158497436


"\n#making graph of simulations using a number of point pars up to user bound\ndistance_approxes = [find_ave_distance(num_p_pairs) for num_p_pairs in range(1,num_point_pairs+1)]\nplt.plot(np.arange(num_point_pairs),distance_approxes)\nplt.title('Average Distance in Unit Circle Simulations')\nplt.xlabel('# of points used in simulation')\nplt.ylabel('Average Distance')\nplt.show()\n"

10 million point simulation answer of  0.909 is incredibly close to the analytical answer of 0.905

